src/pkg/lint/validate_schema.go

```diff
@@ -0,0 +1,58 @@
+package lint
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+const (
+	zarfInvalidPrefix = "zarf.yaml is not valid:"
+	zarfWarningPrefix = "zarf schema warning:"
+	ZarfTemplateVar   = "###ZARF_PKG_TMPL_"
+)
+
+func validateSchema(unmarshalledYaml interface{}, jsonSchema []byte) error {
+	schemaLoader := gojsonschema.NewBytesLoader(jsonSchema)
+	documentLoader := gojsonschema.NewGoLoader(unmarshalledYaml)
+
+	result, err := gojsonschema.Validate(schemaLoader, documentLoader)
+	if err != nil {
+		return err
+	}
+
+	if !result.Valid() {
+		errorMessage := zarfInvalidPrefix
+		for _, desc := range result.Errors() {
+			errorMessage = fmt.Sprintf("%s\n - %s", errorMessage, desc.String())
+		}
+		err = errors.New(errorMessage)
+	}
+
+	return err
+}
+
+func checkForVarInComponentImport(zarfYaml types.ZarfPackage) error {
+	valid := true
+	errorMessage := zarfWarningPrefix
+	componentWarningStart := "component."
+	for i, component := range zarfYaml.Components {
+		if strings.Contains(component.Import.Path, ZarfTemplateVar) {
```

Comment:
each of these are rules that could not be enforced in the schema due to many reasons.